controls/SV-244531.rb

```diff
@@ -39,16 +34,14 @@
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
 
-  non_interactive_shells = input('non_interactive_shells')
-
-  ignore_shells = non_interactive_shells.join('|')
+  ignore_shells = input('non_interactive_shells').join('|')
+  exempt_home_users = input('exempt_home_users').join('|')
 
   findings = Set[]
-  users.where { !shell.match(ignore_shells) && (uid >= 1000 || uid == 0) }.entries.each do |user_info|
-    findings += command("find #{user_info.home} -xdev -not -name '.*' -perm /027").stdout.split("\n")
+  users.where { !username.match(exempt_home_users) && !shell.match(ignore_shells) && (uid >= 1000 || uid.zero?) }.entries.each do |user_info|
+    findings += command("find #{user_info.home} -xdev -not -name '.*' -perm /027 -type f").stdout.split("\n")
   end
   describe findings do
     it { should be_empty }
   end
```

Comment:
update this control to use the 'more_permissive_than' matcher - I spent a lot of time writting that :)